### PR TITLE
added check for duplicate extensions

### PR DESCRIPTION
**Pull request by SeanKuehl**
Added a basic check for duplicate extensions to prevent the duplicate "png" issue#131 previously reported.

### CODE REVIEW

**Comment by dbartolini** [comment]
This does not prevent duplicated entries in `_all_extensions_importer_data.extensions`, just avoids adding duplicated entries from `data`. For example, the following code will still produce duplicates:

    register_importer("Sprite", { "png" }, SpriteResource.import, 0.0);
    register_importer("Texture", { "png", "tga", "dds", "ktx", "pvr" }, TextureResource.import, 2.0);

**Comment by SeanKuehl** [comment]
I see the register_importer function. I only did the register_importer_internal. If I add a similar approach to register_importer will this fix the issue? I would test this myself but I've had issues trying to build Crown myself. I've asked a question in the Discord but haven't heard back yet.

File: tools/level_editor/project.vala [hunk 1]
```diff
@@ -551,7 +551,18 @@ public class Project
...
+		foreach (string ext in data.extensions){
+			if (uniqueExtensions.contains(ext)){
+				//it's a duplicate, don't add it
+			}
+			else {
+				uniqueExtensions.add(ext);
+			}
+		}
+
+		_all_extensions_importer_data.extensions.add_all(uniqueExtensions);
```
Comment (dbartolini) [inline tools/level_editor/project.vala:565]: You can throw `extensions.add_all()` away and fill the `extensions` array manually with a loop similar to the one you proposed.

**Comment by SeanKuehl** [comment]
I've changed the loop to check for duplicates in all_extensions and add only unique ones.

**Pull request merged**